attribute/value.go

```diff
@@ -113,7 +137,30 @@ func StringValue(v string) Value {
 
 // StringSliceValue creates a STRINGSLICE Value.
 func StringSliceValue(v []string) Value {
-	return Value{vtype: STRINGSLICE, slice: attribute.StringSliceValue(v)}
+	return Value{vtype: STRINGSLICE, iface: attribute.StringSliceValue(v)}
+}
+
+// SliceValue creates a SLICE Value.
+func SliceValue(v []Value) Value {
+	var zero Value
+	cp := reflect.New(reflect.ArrayOf(len(v), reflect.TypeOf(zero))).Elem()
+	reflect.Copy(cp, reflect.ValueOf(v))
+	return Value{vtype: SLICE, iface: cp.Interface()}
+}
+
+// MapValue creates a MAP Value.
+// v is sorted by key.
```

Comment:
We should also deduplicate keys, but this is an implementation detail not relevant for the prototype.